### ISSUE TITLE
Masking Patches Question

### ISSUE THREAD

**Issue by druvpatel** [lucidrains/vit-pytorch#26]
Hi! Hope you're doing well and thanks for the great resource. 

Many of my images have padding-patches, so I am trying to use the mask input to have the model ignore these padding patches. It seems I'm encountering an error that when I use these masks, the output of my model becomes [nan, nan] (for a 2-class output problem).

I believe this may be because in the` masked_fill_` (line 59 in vit_pytorch.py),the patch indices in the mask I want to ignore (which I've set to False in the mask) gets set to `float(-inf)` in `dots`. This then gets softmaxed, resulting in softmax(-inf) -> `nan`.

As a fix, I've added this after line 62 in vit_pytorch.py: `attn[torch.isnan(attn)] = float(0)`

I'm relatively new to attention, and was wondering if this is a correct way to approach this.

Thanks!

**Comment by lucidrains** [comment]
@druvpatel yes indeed, i made the masking code in haste

released the better version! https://github.com/lucidrains/vit-pytorch/commit/6c8dfc185ea41f4d2388e4d33bbb76f900ff8a0a

**Comment by druvpatel** [comment]
Thank you so much! 😄
